topcoffea/modules/YieldTools.py

```diff
@@ -0,0 +1,436 @@
+import gzip
+import json
+import pickle
+import numpy as np
+from coffea import hist
+#from topcoffea.modules.HistEFT import HistEFT
+from topcoffea.modules.paths import topcoffea_path
+
+class YieldTools():
+
+    def __init__(self):
+
+        # The path to the file listing the luminosity for each year
+        self.LUMI_FILE = topcoffea_path("json/lumi.json")
+
+        # The order of the categories in the TOP-19-001 AN yield tables
+        self.CAT_LST = ["cat_2lss_p","cat_2lss_m","cat_3l_1b_offZ_p","cat_3l_1b_offZ_m","cat_3l_2b_offZ_p","cat_3l_2b_offZ_m","cat_3l_1b_onZ","cat_3l_2b_onZ","cat_4l"]
+
+        # A dictionary mapping names of samples in the samples axis to a short version of the name
+        self.PROC_MAP = {
```

Comment:
I've been working on a way to dynamically create these. In the datacard maker I have a dictionary called `rename` (though you don't have to call yours this), which is initially defined for things like `ttZ` to `ttll`
https://github.com/TopEFT/topcoffea/blob/ht_cuts/analysis/topEFT/datacard_maker.py#L17
then I dynamically grab the process names from a histogram, and use a regex to extract the important parts, and append this to `rename`
https://github.com/TopEFT/topcoffea/blob/ht_cuts/analysis/topEFT/datacard_maker.py#L41-L42
which produces:
`{'tZq': 'tllq', 'tllq_privateUL17': 'tllq', 'ttZ': 'ttll', 'ttlnu': 'ttlnuJet_privateUL17', 'tttt': 'tttt_privateUL17', 'ttll': 'ttllJet_privateUL17', 'tllq': 'tllq_privateUL17', 'ttH': 'ttHJet_privateUL17', 'tHq': 'tHq_privateUL17'}`
If you're interested, I can help you implement something similar here, so we don't have to keep track of all the different names floating around. The only issues I foresee with this is we must conform to the regex `process_label`, or rewrite the regex.